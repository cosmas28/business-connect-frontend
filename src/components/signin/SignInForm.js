// ./src/components/sigin/SignInForm.js
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

/**
 *
 * @param {Object} props - properties from parent component
 * @returns {JSX} - render dashboard title bar
 */
const SignInForm = props => {

    // display flash message if it exists
    let responseMessage = null;
    if (props.statusCode === 200) {
        responseMessage = <div className="alert alert-success alert-dismissible fade show" role="alert">
            {props.loginMessage}
            <button type="button" className="close" data-dismiss="alert" aria-label="Close">
                <span aria-hidden="true">&times;</span>
            </button>
        </div>;
    } else {
        responseMessage = <div className="alert alert-danger alert-dismissible fade show" role="alert">
            <strong>Error</strong> {props.loginMessage}
            <button type="button" className="close" data-dismiss="alert" aria-label="Close">
                <span aria-hidden="true">&times;</span>
            </button>
        </div>;
    }

    return (
        <div className="main-login-page">
            <p>Sign in to your account</p>
            {props.loginMessage &&
                responseMessage
            }
            <form onSubmit={props.handleLoginSubmitForm}>
                <div className="form-input-division">
                    <input type="email"
                        name="email"
                        className="form-text-input form-control"
                        placeholder="Email"
                        onChange={props.handleLoginInputChange}
                        required
                    />
                </div>
                <div className="form-input-division">
                    <input type="password"
                        id="password"
                        name="password"
                        className="test form-text-input form-control"
                        placeholder="Password"
                        onChange={props.handleLoginInputChange}
                        required
                    />
                </div>
                <div className="form-input-division">
                    <button type="submit" className="btn btn-primary input-default form-btn">Sign In</button>
                </div>
                <div className="form-input-division">
                    <Link className="nav-link" to="/reset_password/confirm_email">Forgot password?</Link>
                </div>
                <hr />
                <div className="form-input-division">
                    <Link className="nav-link" to="/">You don't have an account?</Link>
                </div>
            </form>
        </div>
    );
};

// define prop types for SignInForm
SignInForm.propTypes = {
    'handleLoginInputChange': PropTypes.func.isRequired,
    'handleLoginSubmitForm': PropTypes.func.isRequired,
    'loginMessage': PropTypes.string,
    'statusCode': PropTypes.number
};

export default SignInForm;
