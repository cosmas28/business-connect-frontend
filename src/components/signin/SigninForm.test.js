import React from 'react';
import SignInForm from './SigninForm';

describe('Sign in form test', () => {
    let mountedSignInForm = null,
       props = null;
    const mockHandleSubmitForm = jest.fn(),
        mockInputChange = jest.fn();
    const component = () => {
        if (!mountedSignInForm) {
            mountedSignInForm = mount(
                <SignInForm {...props} />
            );
        }

        return mountedSignInForm;
    };

    beforeEach(() => {
        props = {
            'handleLoginInputChange': mockHandleSubmitForm,
            'handleLoginSubmitForm': mockInputChange,
            'loginPutErrorMessage': null,
            'loginSuccessMessage': null
        };
        mountedSignInForm = null;
    });

    it('renders and matches our snapshot', () => {
        expect(component()).toMatchSnapshot();
    });

    it('contains the form', () => {
        expect(component().find('input')).toHaveLength(2);
        expect(component().find('button')).toHaveLength(1);
    });

    it('should call `handleLoginInputChange` when input field is changed', () => {
        component().find('input.test').simulate('change');
        expect(props.handleLoginInputChange).toBeCalled();
    });

    it('should call `handleLoginSubmitForm` when submit button is called', () => {
        component().find('form').simulate('submit');
        expect(props.handleLoginSubmitForm).toBeCalled();
    });

});
