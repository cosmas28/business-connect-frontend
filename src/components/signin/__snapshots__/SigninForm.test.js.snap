// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign in form test renders and matches our snapshot 1`] = `
<SignInForm
  handleLoginInputChange={[MockFunction]}
  handleLoginSubmitForm={[MockFunction]}
  loginPutErrorMessage={null}
  loginSuccessMessage={null}
>
  <div
    className="main-login-page"
  >
    <h2>
      Sign in to your account
    </h2>
    <form
      onSubmit={[MockFunction]}
    >
      <div
        className="form-input-division"
      >
        <input
          className="input-default form-text-input"
          name="username"
          onChange={[MockFunction]}
          placeholder="Username"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-input-division"
      >
        <input
          className="test input-default form-text-input"
          id="password"
          name="password"
          onChange={[MockFunction]}
          placeholder="Password"
          required={true}
          type="password"
        />
      </div>
      <div
        className="form-input-division"
      >
        <button
          className="btn btn-primary input-default form-btn"
          type="submit"
        >
          Sign In
        </button>
      </div>
      <div
        className="form-input-division"
      >
        <a
          href=""
        >
          Forgot password?
        </a>
      </div>
      <hr />
      <div
        className="form-input-division"
      >
        <a
          href=""
        >
          You don't have an account? 
        </a>
      </div>
    </form>
  </div>
</SignInForm>
`;
